### ISSUE TITLE
custom loss for many classes classification

### ISSUE THREAD

**Issue by wyfSunflower** [Microsoft/LightGBM#2084]
Is it possible to define a custom loss function for many class classification?
I want to use lightgbm to construct a many class classification model for indoor bluetooth localization,I cut target area into many pieces with different id(hundreds of  class), Input is some signal strength values at each piece, output is the id of that piece, I want to make loss larger when distance error of predict id and target id is bigger than a specified value. current is multi-logloss, it does not work well when class number too big

**Comment by guolinke** [comment]
refer to https://github.com/Microsoft/LightGBM/issues/1308
